url78f6619d18_IntersectAustralia_rifcs_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionJ8/src/RIFCSWrapper.java

```diff
@@ -0,0 +1,203 @@
+/*     */     {
+/*  90 */       throw new RIFCSException(pce);
+/*     */     }
+/*     */   }
+/*     */ 
+/*     */   public RIFCSWrapper(Document d)
+/*     */     throws RIFCSException
+/*     */   {
+/* 105 */     this.doc = d;
+/* 106 */     this.rifcs = new RIFCS(d);
+/*     */   }
+/*     */ 
+/*     */   public Document getRIFCSDocument()
+/*     */   {
+/* 118 */     return this.doc;
+/*     */   }
+/*     */ 
+/*     */   public RIFCS getRIFCSObject()
+/*     */   {
+/* 130 */     return this.rifcs;
+/*     */   }
+/*     */ 
+/*     */   public void write(OutputStream os)
+/*     */   {
+/* 142 */     DOMImplementation impl = this.doc.getImplementation();
+/* 143 */     DOMImplementationLS implLS = (DOMImplementationLS)impl.getFeature("LS", "3.0");
+/*     */ 
+/* 145 */     LSOutput lso = implLS.createLSOutput();
+/* 146 */     lso.setByteStream(os);
+/* 147 */     LSSerializer writer = implLS.createLSSerializer();
+/* 148 */     DOMConfiguration domConfig = writer.getDomConfig();
+/* 149 */     domConfig.setParameter("format-pretty-print", Boolean.valueOf(true));
+/* 150 */     writer.write(this.doc, lso);
+/*     */   }
+/*     */ 
+/*     */   public String toString()
+/*     */   {
+/*     */     try
+/*     */     {
+/* 164 */       TransformerFactory factory = TransformerFactory.newInstance();
+/* 165 */       Transformer transformer = factory.newTransformer();
+/* 166 */       StringWriter writer = new StringWriter();
+/* 167 */       Result result = new StreamResult(writer);
+/* 168 */       Source source = new DOMSource(this.doc);
+/* 169 */       transformer.transform(source, result);
+/* 170 */       writer.close();
+/* 171 */       return writer.toString();
+/*     */     }
+/*     */     catch (TransformerException te)
+/*     */     {
+/* 175 */       return null;
+/*     */     }
+/*     */     catch (IOException ioe) {
+/*     */     }
+/* 179 */     return null;
+/*     */   }
+/*     */ 
+/*     */   public void validate()
+/*     */     throws SAXException, MalformedURLException, IOException, ParserConfigurationException
+/*     */   {
+/* 205 */     SchemaFactory factory = SchemaFactory.newInstance("http://www.w3.org/2001/XMLSchema");
+/*     */ 
+/* 207 */     Schema schema = factory.newSchema(doXercesWorkaround());
+/*     */ 
+/* 210 */     Validator validator = schema.newValidator();
+/* 211 */     validator.validate(new DOMSource(this.doc));
+/*     */   }
+/*     */ 
+/*     */   private Source doXercesWorkaround()
+/*     */     throws SAXException, MalformedURLException, IOException, ParserConfigurationException
+/*     */   {
+/* 218 */     DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+/* 219 */     factory.setNamespaceAware(true);
+/* 220 */     DocumentBuilder builder = factory.newDocumentBuilder();
+/*     */ 
+/* 222 */     Document docRO = builder.parse(new URL("http://services.ands.org.au/home/orca/schemata/registryObjects.xsd").openStream());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *openStream*. Currently, there are execution paths that do not contain closure statements, for example, when DocumentBuilder.parse throws an exception. Ensure the object returned by *openStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).